### PR TITLE
#23 | 08 create health metrics monitoring lecture

### PR DESCRIPTION
**Pull request by grygy**

### CODE REVIEW

File: 08/passenger-service/src/main/java/cz/muni/fi/airportmanager/passengerservice/client/BaggageClientResource.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+@RegisterRestClient(configKey = "baggage-resource")
+public interface BaggageClientResource {
+
+    @GET
+    @Path("/baggage/passenger/{passengerId}")
+    Uni<List<Baggage>> getBaggageForPassengerId(@PathParam("passengerId") Long passengerId);
+
+    @GET
+    @Path("/q/health/ready")
+    Uni<String> readinessCheck();
```
Comment (xstefank) [inline 08/passenger-service/src/main/java/cz/muni/fi/airportmanager/passengerservice/client/BaggageClientResource.java:24]: Maybe use SmallRyeHealth class here

File: 08/passenger-service/src/main/java/cz/muni/fi/airportmanager/passengerservice/repository/PassengerRepository.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    public Uni<List<Passenger>> findPassengersForFlight(Long flightId) {
+        return find("flightId", flightId).list();
+    }
+
+    /**
+     * Get list of hydrated notifications
+     *
+     * @return list of hydrated notifications with passenger email
+     */
+    public Uni<List<NotificationDto>> findHydratedNotifications() {
```
Comment (xstefank) [inline 08/passenger-service/src/main/java/cz/muni/fi/airportmanager/passengerservice/repository/PassengerRepository.java:71]: no idea what hydrated means in this context

File: 08/passenger-service/src/test/java/cz/muni/fi/airportmanager/passengerservice/resource/PassengerResourceTestIT.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.is;
+
+@QuarkusIntegrationTest
+@TestHTTPEndpoint(PassengerResource.class)
+class PassengerResourceTestIT {
```
Comment (xstefank) [inline 08/passenger-service/src/test/java/cz/muni/fi/airportmanager/passengerservice/resource/PassengerResourceTestIT.java:17]: Why this isn't in the PassengerResourceTest?

File: 08/baggage-service/src/main/resources/application.properties [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+
+# drop and create the database at startup (use `update` to only update the schema)
+quarkus.hibernate-orm.database.generation=drop-and-create
+
+# include swagger in prod
+quarkus.swagger-ui.always-include=true
+
+# kafka
+mp.messaging.outgoing.baggage-state-change.value.serializer=io.quarkus.kafka.client.serialization.ObjectMapperSerializer
```
[the remark this answers is not in the document]
Comment (grygy) [reply 08/baggage-service/src/main/resources/application.properties:21]: In this version, it is needed. Tired without it. After the upgrade, I'll remove it.

File: 08/baggage-service/src/main/java/cz/muni/fi/airportmanager/baggageservice/model/example/Examples.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
+package cz.muni.fi.airportmanager.baggageservice.model.example;
+
+
+public class Examples {
```
[the remark this answers is not in the document]
Comment (grygy) [reply 08/baggage-service/src/main/java/cz/muni/fi/airportmanager/baggageservice/model/example/Examples.java:4]: they are used for open api `examples = @ExampleObject(name = "baggage", value = Examples.VALID_BAGGAGE_LIST)`

File: 08/baggage-service/src/main/java/cz/muni/fi/airportmanager/baggageservice/resources/BaggageResource.java [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+    )
+    public Uni<RestResponse<Baggage>> get(long id) {
+        return baggageService.getBaggage(id)
+                .onItem().transform(baggage -> RestResponse.status(Response.Status.OK, baggage))
+                .onFailure(IllegalArgumentException.class).recoverWithItem(RestResponse.status(Response.Status.NOT_FOUND));
+    }
+
+
+    /**
+     * Helper method for to delete all baggage
```
[the remark this answers is not in the document]
Comment (grygy) [reply 08/baggage-service/src/main/java/cz/muni/fi/airportmanager/baggageservice/resources/BaggageResource.java:128]: removed

File: 08/baggage-service/src/main/resources/application.properties [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+quarkus.http.test-port=8087
+
+%prod.quarkus.http.port=8080
+
+# configure your datasource
+%prod.quarkus.datasource.db-kind=postgresql
+# Inserted from docker-compose
+%prod.quarkus.datasource.username=${QUARKUS_DATASOURCE_USERNAME}
+%prod.quarkus.datasource.password=${QUARKUS_DATASOURCE_PASSWORD}
+%prod.quarkus.datasource.reactive.url=${QUARKUS_DATASOURCE_REACTIVE_URL}
```
[the remark this answers is not in the document]
Comment (grygy) [reply 08/baggage-service/src/main/resources/application.properties:11]: I've used them as a part of task in 06 for students to know what kind of variables they need to setup and to which key they relate.

File: 08/baggage-service/src/main/java/cz/muni/fi/airportmanager/baggageservice/resources/BaggageResource.java [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+
+    /**
+     * Create a new baggage
+     *
+     * @param baggage baggage to create.
+     * @return created baggage
+     */
+    @POST
+    @Produces(APPLICATION_JSON)
+    @Consumes(APPLICATION_JSON)
```
[the remark this answers is not in the document]
Comment (grygy) [reply 08/baggage-service/src/main/java/cz/muni/fi/airportmanager/baggageservice/resources/BaggageResource.java:74]: In the open API exercise i wanted to show this explicitly.

File: 08/passenger-service/src/main/java/cz/muni/fi/airportmanager/passengerservice/repository/PassengerRepository.java (hunk 2 above)
Comment (grygy) [reply 08/passenger-service/src/main/java/cz/muni/fi/airportmanager/passengerservice/repository/PassengerRepository.java:71]: renmaed

File: 08/passenger-service/src/test/java/cz/muni/fi/airportmanager/passengerservice/resource/PassengerResourceTestIT.java (hunk 3 above)
Comment (grygy) [reply 08/passenger-service/src/test/java/cz/muni/fi/airportmanager/passengerservice/resource/PassengerResourceTestIT.java:17]: Tests are focused on the whole service without any isolation

**Pull request merged**